src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -288,9 +288,13 @@ trait LowPriorityConfigConvertImplicits {
   }
 
   implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = {
-    val fromString: String => FiniteDuration = (s: String) => DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]]).get match {
-      case v if v.isFinite() => Duration(v.length, v.unit)
-      case _ => throw new IllegalArgumentException(s"Couldn't parse '$s' into a finite duration because it's infinite.")
+    val fromString: String => FiniteDuration = { (s: String) =>
+      DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]])
+        .flatMap {
+          case d: FiniteDuration => Success(d)
+          case _ => Failure(new IllegalArgumentException(s"Couldn't parse '$s' into a FiniteDuration because it's infinite."))
+        }
+        .get
```

Comment:
@melrief & @jcazevedo 
Do you guys thing we should have both safe and unsafe versions of the API? If so, do you want to:
1) Keep the existing methods as is and introduce the new safe versions with the `Safe` suffix
2) Rename the existing to have an `Unsafe` suffix and introduce `Safe` method
3) Rename the existing to `Unsafe` and introduce safe versions with the names of the existing methods?

I think my preference among those options is 3, 1, 2.